### ISSUE TITLE
Unable to type onSelect prop for Select correctly

### ISSUE THREAD

**Issue by Markus-ipse** [ant-design/ant-design#34201]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/bold-lamarr-zze2rw?file=/src/App.tsx)

### Steps to reproduce
Add an onSelect prop to a <Select /> when using typescript. 

### What is expected?
That I am able to get proper types for the onSelect callback parameters.

### What is actually happening?
both arguments get implicitly typed as any, even when trying to use the generics syntax `<Select<ValueType, OptionType>></Select`, this syntax does however seem to apply to the filterOption callback which does get its params typed based on ValueType and OptionType. 

| Environment | Info |
|---|---|
| antd | 4.18.8 |
| React | react 17.0.2 |
| System | windows 10 |
| Browser | n/a |

**Comment by archiekd** [comment]
> * [x]  I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.
> 
> ### Reproduction link
> [![Edit on CodeSandbox](https://camo.githubusercontent.com/90808661433696bc57dce8d4ad732307b5cec6270e6b846f114dcd7ee7f9458a/68747470733a2f2f636f646573616e64626f782e696f2f7374617469632f696d672f706c61792d636f646573616e64626f782e737667)](https://codesandbox.io/s/bold-lamarr-zze2rw?file=/src/App.tsx)
> 
> ### Steps to reproduce
> Add an onSelect prop to a `<Select />` when using typescript.
> 
> ### What is expected?
> That I am able to get proper types for the onSelect callback parameters.
> 
> ### What is actually happening?
> both arguments get implicitly typed as any, even when trying to use the generics syntax `<Select<ValueType, OptionType>></Select>`, this syntax does however seem to apply to the filterOption callback which does get its params typed based on ValueType and OptionType.
> 
> Environment	Info
> antd	4.18.8
> React	react 17.0.2
> System	windows 10
> Browser	n/a

Yeah I am having the same issue some point after 4.17.1 the types have changed and no longer allow inference from the value or options types

**Comment by DBSDs** [comment]
can i work on this?

**Comment by afc163** [comment]
@DBSDs sure

**Comment by DBSDs** [comment]
> @DBSDs sure

谢谢，我认为他的pr可能已经修复了这个问题[#729](https://github.com/react-component/select/pull/729#issue-1174205238)，并且与之前的commit并不是回退的关系。留言中上一个commit的代码是为onSelect添加泛型，pr的代码是将先前定义的onSelect类型删除。

不过可能需要在用户没有为Select泛型传参的时候，定义一个初始的ValueType，`SelectProps<ValueType = any, OptionType extends BaseOptionType = DefaultOptionType>`改成`SelectProps<ValueType = DraftValueType, OptionType extends BaseOptionType = DefaultOptionType>`不知道是否可行？
